### ISSUE TITLE
Hard hang in find_python_version()

### ISSUE THREAD

**Issue by mathrick** [sarugaku/pythonfinder#107]
I'm not sure  what's causing this, and what details of my environment might be relevant, but I'm encountering a *hard* hang when calling `Finder.find_python_version` or `Finder.find_all_python_versions`. The Python process becomes entirely unresponsive, ignoring Ctrl+C, Ctrl+Z and the like. Killing it from another terminal is the only way to quit.

Environment:
* Ubuntu 20.04
* Python 3.x (tried with 3.6.8 and 3.8.5)
* Using `pyenv` with `pyenv-virtualenv`
* Seems to affect all virtualenvs equally

**Comment by blueyed** [comment]
Try <kbd>Ctrl-d</kbd>.

I am seeing this with pyrepl being installed (via pdb++), which ships `pythoni`, which gets picked up / tried by pythonfinder, but acts as an interactive shell always (ignoring `--version` etc).

I think pythonfinder maybe should close stdin when trying out existing python interpreters maybe at least.

**Comment by mathrick** [comment]
@blueyed: you're right, I do have `pdb++` installed. Ctrl+D didn't help (I'm assuming because `pythonfinder` doesn't attach the TTY when spawning subprocesses), but that sounds very likely.

**Comment by mathrick** [comment]
Now that you mention it, I see `pythoni` in the `ps` output, so you're definitely right.

**Comment by blueyed** [comment]
You might need it several times: press it and wait a few seconds, before pressing it again (to not accidentally close the terminal).

**Comment by matteius** [comment]
https://github.com/sarugaku/pythonfinder/pull/162
